### ISSUE TITLE
Automatically lock old closed issues?

### ISSUE THREAD

**Issue by icemac** [zopefoundation/meta#7]
Today I stumbled across the lock bot app: https://github.com/apps/lock

It locks closed issues after a defined amount of time of there is no further activity. It can encourage users to create a new issue if there is a problem related to the closed issue.

What do you think about this app?
Could it be helpful for (at least some) zopefoundation repositories? 
Do you see a benefit or would it only add more noise?

**Comment by mgedmin** [comment]
Is that the thing that's been annoying me lately by reminding me about old pip issues that never got closed properly or something?

I'm not a fan.

**Comment by jugmac00** [comment]
It is not like we have hundreds or thousands of open issues. So I think it is better when some human looks at old issues from time to time and either closes them manually as they are maybe already fixed or reminds the issue starter to either provide additional information or the issue gets closed.

P.S.: Just last week I had a ping pong match with a mechanism on the Fedora issue tracker which closed my ticket erroneously every time I tried to reopen it - which was the proposed solution :-(
P.P.S.: I lost the ping pong match and the bug remains open - with no corresponding issue.

**Comment by d-maurer** [comment]
Jürgen Gmach wrote at 2019-6-3 02:53 -0700:
>It is not like we have hundreds or thousands of open issues. So I think it is better when some human looks at old issues from time to time and either closes them manually as they are maybe already fixed or reminds the issue starter to either provide additional information or the issue gets closed.

When I understand this right, then this discussion is not about
old *open* issues but about old *closed* issues.
"locking" those issues likely means, one can no longer add comments to
them -- and is forced to open a new issues when one is not satified
with the old "solution".

**Comment by dataflake** [comment]
As Michael said, this is about *closed* issues.

I like the idea, but as with `black`, my question is: Do we have a problem that is solved by this tool? If not then we're just adding yet another configuration file to repositories without clear benefit.

I don't have the impression that people picking up on closed issues is a problem for any of our repositories.

**Comment by jugmac00** [comment]
Thanks @d-maurer - I misread the text completely.

I see no reason for the usage of such a process.

On the contrary, this would prevent maybe valuable information, like issue was not fixed 
(in all cases), or this introduced a bug in Plone, or just somebody wants to say thanks for a fix or a feature.

**Comment by icemac** [comment]
Okay, as no-one seems to see a benefit in locking closed issues automatically, I am closing this issue (without locking it 😉 )

**Comment by thinkyhead** [comment]
Thanks for this. It's just what I'm looking for… I'd rather not lock issues immediately as they are closed.

**Comment by dlenski** [comment]
If you don't want to lock issues _automatically_, but do want to clean them up en masse from time to time … consider https://github.com/dlenski/github-lockify? (I recently wrote it to manage the issues for a couple repos where users keep commenting on irrelevant- or barely-relevant old issues, rather than opening new ones.)

It allows you to select a subset of issues by age/labels/assignee/creator, and lock those that match:

```sh
$ github-lockify --help
usage: github-lockify [-h] [--do-it] [-r {off-topic,too heated,resolved,spam}]
                      [-u USER] [-t TOKEN] [-U DAYS] [-C DAYS]
                      [--created-age DAYS] [-l LABEL] [-a USERNAME]
                      [-c USERNAME]
                      owner repo

Lock closed GitHub issues en masse.

positional arguments:
  owner                 GitHub repository owner
  repo                  Github repository name

optional arguments:
  -h, --help            show this help message and exit
  --do-it               Actually lock the issues (default is a dry-run where
                        issues are simply listed).
  -r {off-topic,too heated,resolved,spam}, --lock-reason {off-topic,too heated,resolved,spam}
                        Lock-reason to apply (default resolved)

Authentication:
  Not required for a dry-run, but required to actually lock issues.

  -u USER, --user USER  GitHub username (default is same as repository owner)
  -t TOKEN, --token TOKEN
                        GitHub personal access token with repo scope (see
                        https://github.com/settings/tokens).

Selecting closed issues to lock:
  All criteria must match in order for an issue to be locked.

  -U DAYS, --updated-age DAYS
                        Only lock if UPDATED at least this many days ago
  -C DAYS, --closed-age DAYS
                        Only lock if CLOSED at least this many days ago
  --created-age DAYS    Only lock if CREATED at least this many days ago
  -l LABEL, --label LABEL
                        Only lock if this label is applied (may be specified
                        repeatedly to require multiple labels)
  -a USERNAME, --assignee USERNAME
                        Only lock if assigned to this user ("none" for
                        unassigned)
  -c USERNAME, --creator USERNAME
                        Only lock if created by this user
```

**Comment by d-maurer** [comment]
> If you don't want to lock issues _automatically_, but do want to clean them up en masse

If "cleaning them up" means "closing them", I would not agree. I filed (e.g.) several issues for `Products..PageTemplates` which are actually `chameleon` issues (i.e. their fix is out of control by the Zope contributors). In my view, those issue reports should be easily found via searches (which likely means they should remain open) until they are resolved in `chameleon`; this can help people to avoid the long analysis effort I have invested to understand the related observations.
